### PR TITLE
adds affiliation for job board

### LINKED ISSUE #401: Add "IncBuilders Partnership Opportunity" to job board
We want to be able to differentiate positions that are posted through our partnership with IncBuilders from other job postings.

Let's add a checkbox on the job posting form, with the text "This is an IncBuilders Partnership Opportunity".  If checked, let's display the text "IncBuilders Partnership Opportunity" on the posting and in the list view, maybe below the position and company name?  Or maybe, add it to the position field?

IncBuilders has requested the ability to submit postings via API, so they may integrate this with their platform. After looking at the view, it should theoretically be possible to POST to this endpoint as-is, with the caveat that we'd need to work out authentication.  I haven't really looked at the login_required decorator in a while, but maybe we could use basic-auth and an authentication header?

### PR DESCRIPTION
**Pull request by elmq0022**
Fixes #401.  @eevelweezel please review.

### CODE REVIEW

**Comment by nickymarino** [comment]
@elmq0022 can you change the description from `For #401` to `Fixes #401`? That way GitHub will automatically link the PR to the issue [(reference)](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: chipy_org/apps/job_board/models.py [hunk 1]
```diff
@@ -41,6 +41,14 @@ def get_queryset(self):
         )
 
 
+class Affiliation(CommonModel):
+    description = models.CharField(max_length=MAX_LENGTH)
```
Comment (nickymarino) [inline chipy_org/apps/job_board/models.py:45]: What's the description for here, if it's not what's going to be displayed on the job page? (Currently the viewable description is handled by [this](https://github.com/chicagopython/chipy.org/pull/412/files#diff-b3364321a2ae5d2c87a3fa9356b5a74e64c3eae94254dad103d1d0e5ec5a8b2dR42))

Should the affiliation be tied to some user that has basic auth access to the job creation API endpoint?

**Review by nickymarino — COMMENTED** [review]
What's a good way to test this functionality? I tried creating an `Affiliation` and `Job Post` in the /admin view, but now it's stuck in "Submitted" (instead of "Approved"). Where do I go to approve a job post so that it shows up on the job board?

I also had a few questions about the `Affiliation` data model that I left a comment on.

I put a few questions about the data model in the review.

File: chipy_org/apps/job_board/models.py (hunk 1 above)
Comment (elmq0022) [reply chipy_org/apps/job_board/models.py:45]: You can change the status of the job (submitted, approved, rejected) through the admin.  Description is used in the template be default - django tries hard to get a string representation of and object and I had set the __str__ method to return description.  I'll update it though just to make it clear though.

Comment (nickymarino) [reply chipy_org/apps/job_board/models.py:45]: Oh duh of course, thanks for clarifying those 😄

**Review by nickymarino — APPROVED** [review]
Looks good! 🦞

**Review by sophm256 — CHANGES_REQUESTED** [review]
Thanks for working on this PR! Very nice job overall :smile: 

There are a few changes I would like to suggest.

File: chipy_org/apps/job_board/forms.py [hunk 2]
```diff
@@ -21,12 +21,17 @@ class Meta:
...
             "is_sponsor",
+            "affiliation",
             "can_host_meeting",
             "company_website",
             "how_to_apply",
             "agree_to_terms",
         ]
 
+        help_texts = {
+            "affiliation": ("Let us know if this job posting is associated with a third party."),
```
Comment (sophm256) [inline chipy_org/apps/job_board/forms.py:32]: ### Currently, this is what the affiliation field dropdown looks like when you go to create a job post:
Screen Shot 2021-05-02 at 11 26 25 AM

### I would like for it to look like this instead:
Screen Shot 2021-05-02 at 12 02 43 PM

To achieve this, please make 2 changes.

STEP 1:
In `job_board/forms.py`, line 32, change the text to: 
"Is this job post affiliated with a 3rd party? Please select:"

STEP 2:
In `job_board/templates/job_post_form.html`, within the `<table> </table>` tags, and right before the `else` clause, insert this code.
```
{% elif field == job_post_form.affiliation %}
                <tr>
                    <td colspan="2">
                        {{field.help_text}} {{field}}
                        {{ field.errors }}
                    </td>
                </tr> 
```

File: chipy_org/apps/job_board/templates/job_board/job_post_detail.html [hunk 3]
```diff
@@ -38,5 +38,8 @@ <h2>
 
 <b>How To Apply:</b> {{ job_post.how_to_apply|linebreaks }}
 
+{% if job_post.affiliation %}
+    This posting is affiliated with <a href="{{job_post.affiliation.url}}">{{ job_post.affiliation.description }}</a>
```
Comment (sophm256) [inline chipy_org/apps/job_board/templates/job_board/job_post_detail.html:42]: Line 42, please bold the text "This posting is affiliated with"

**Comment by sophm256** [comment]
I'd like to have the 'Builders Inc Opportunity' to show up in the listview.

Let's assume, we add an `affiliation` object named 'Builders Inc' through the admin panel.

Please add the following code to the `job_board/templates/job_post_list.html`:
```
{% if post.affiliation %}
    <br>{{ post.affiliation }} Opportunity
{% endif %}
```
These lines should go after `<br>{{ post.company_name }}` (line 31 in VS Code)

Screen Shot 2021-05-02 at 12 44 09 PM

**Comment by sophm256** [comment]
CORRECTION: I keep referring to them as Builders Inc, when they are in fact IncBuilders. In that case, the 'affiliation' object in the admin panel should say, 'IncBuilders Partnership'.

**Comment by elmq0022** [comment]
looks like you did most of the work already - want to fork my branch and make the changes?

File: chipy_org/apps/job_board/forms.py (hunk 2 above)
Comment (sophm256) [reply chipy_org/apps/job_board/forms.py:32]: On second thought, it's possible to add the verbose name on the affiliation field on the job post model and that will take care of affiliation texts showing up without having to change the template.

**Comment by sophm256** [comment]
@elmq0022, in my local copy of the chipy repo, I did
 
-`git remote add Aaron <aaron’s GitHub address> `
-`get fetch Aaron`
-`git checkout add-job-post-affiliates`

I then made changes to the files and then did a commit to my local version of `add-job-post-affiliates`. I then pushed this branch to my repo on GitHub. And I’ve created a pull requests against that branch in your repo. Please merge my commits into your repo. Thanks!

**Review by sophm256 — APPROVED** [review]

**Comment by sophm256** [comment]
I've also approved this PR so you should be able to merge your PR into main after you've merged my commit into your forked repo.

**Pull request merged**